tom_nonsidereal_airmass/templatetags/nonsidereal_airmass_extras.py

```diff
@@ -44,3 +48,90 @@ def nonsidereal_target_plan(context):
         'target': context['object'],
         'visibility_graph': visibility_graph
     }
+
+
+@register.inclusion_tag('tom_targets/partials/target_distribution_nonsidereal.html')
+def target_distribution_nonsidereal(targets):
+    """
+    Displays a plot showing on a map the locations of ALL targets in the TOM.
+    Positions are plotted every ten days.
+    """
+    locations = targets.filter(type=Target.SIDEREAL).values_list('ra', 'dec', 'name')
+    data = []
+    for target in targets:
+        if target.type=='NON_SIDEREAL':
+            if target.scheme!='EPHEMERIS':
+                (ra, dec) = get_arc(target)
+                data.append(
+                    dict(
+                        lon=ra,
+                        lat=dec,
+                        text=target.name,
+                        hoverinfo='text',
+                        mode='lines',
+                        type='scattergeo'
+                    ))
+            elif target.scheme=='EPHEMERIS':
```

Comment:
I'm a little wary of putting a dependence on `EPHEMERIS` in this, because `EPHEMERIS` is unique to your TOM. I know you obviously want to use this in your own TOM, though, so I'm trying to figure out a good way to include it as an option for you. Maybe we can extract the construction of the data dictionary for each type into a function, and then the user configures their settings such that a `SCHEME` maps to a function call?

Reply:
I see your point now. Bah. 

I like the idea of a settings trigger, so to speak, but I can't think of the python that would drive that. I could imagine a specific setting like: USE_EPHEMERIS_SCHEME==True

But how does one call a function on condition of EPHEMERIS scheme without asking for that explicitly? Use the negative condition (SCHEME!= all others in normal TOMs)? That's fine until a new scheme is written up.